### PR TITLE
Extend `ListCommand` functionality

### LINKED ISSUE #48: Extend `list` command for viewing classes
As a TA, I want to easily view students from a particular class to get a quick overview.

### PR DESCRIPTION
**Pull request by g-tejas**
Fixes #48 

This allows a user to list all or students from specified class groups.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 1]
```diff
@@ -1,24 +1,73 @@
...
 
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.model.Model;
+import seedu.address.model.person.ListCommandPredicate;
 
 /**
- * Lists all persons in the address book to the user.
+ * Lists students in the address book to the user.
```
Comment (kailashgautham) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:12]: Minor change, but maybe we could say "Lists students in the TA Toolkit"?

**Review by kailashgautham — APPROVED** [review]
Code looks great Tejas! Hopefully I didn't miss anything :D 

Well done!

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 1 above)
Comment (g-tejas) [reply src/main/java/seedu/address/logic/commands/ListCommand.java:12]: Changed!

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        if (!trimmedArgs.isEmpty()) {
+            String[] classGroups = trimmedArgs.split("\\s+");
```
Comment (teojunda) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java:29]: Could I check if it is the intention of the list command to accept multiple classGroup parameters.

If so, could there be a bug if a Person's classGroup contains a space, eg. "F14 3", and user inputs `ls F14 3`, the parser would try to find "F14" and "3" separately.

To solve this, we could either:
1. Enforce that classGroup must not contain whitespace
2. Make the list command: ls c/class group 1 c/class group 2, to distinguish the different classGroups

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 3]
```diff
@@ -1,24 +1,73 @@
...
+        this.predicate = predicate;
+    }
 
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        model.updateFilteredPersonList(predicate);
```
Comment (teojunda) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:37]: Just to check, what is the expected behaviour if ListCommandPredicate is empty?

Comment (teojunda) [reply src/main/java/seedu/address/logic/commands/ListCommand.java:37]: Ah, this is covered in `test()` in ListCommandPredicate.

**Review by teojunda — CHANGES_REQUESTED** [review]
LGTM!
Just want to clarify if there will be any issues if I try to list a classGroup with a whitespace.

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 3 above)
Comment (g-tejas) [reply src/main/java/seedu/address/logic/commands/ListCommand.java:37]: By empty, do you mean the keywords?

**Review by teojunda — APPROVED** [review]
LGTM! We can edit the class group constraints in a separate PR!

File: src/main/java/seedu/address/model/person/ClassGroup.java [hunk 4]
```diff
@@ -4,20 +4,20 @@
...
- * Represents a Person's class/group in the address book.
+ * Represents a Person's class/group in the TA Toolkit.
  * Guarantees: immutable; is valid as declared in {@link #isValidClassGroup(String)}
  */
 public class ClassGroup {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Class/groups should only contain alphanumeric characters, symbols, and spaces. "
-                    + "It cannot start with a symbol and it should not be blank";
+            "Class/groups should only contain alphanumeric characters delimited by a colon.\n"
```
Comment (teojunda) [inline src/main/java/seedu/address/model/person/ClassGroup.java:13]: Should be dash instead of colon right?

**Pull request merged**